sherpa/astro/tests/test_astro_data2.py

```diff
@@ -816,3 +816,46 @@ def test_pha_invalid_units(invalid, make_test_pha):
         pha.units = invalid
 
     assert str(de.value) == f"unknown quantity: '{invalid}'"
+
+
+def test_pha_reverse_sorted():
+    '''This case for a PHA set with reverse sorted channels is a little
+    non-sensical, but it is a very easy way to hit the problem in #1163.
+
+    Really, we want to test that with a larger test that reads real XMM
+    data, which uses reverse energy ordering (see next test).
+
+    So, if thise tests starts to fail, but the next one passes, this one can
+    probably be removed. It's here now because it's fast, easy, and does not
+    require the large test data.
+    '''
+    d = DataPHA('x', np.array([4, 3, 2, 1, 0]), np.array([5, 4, 3, 2, 1]))
+    d.notice(2, 4)
+    assert len(d.get_dep(filter=True)) == 3
+
+
+@pytest.fixture
+def read_test_image(make_data_path):
+    from sherpa.astro.io import read_image
+    filename = 'acisf07999_000N001_r0035_regevt3_srcimg.fits'
+    d = read_image(make_data_path(filename))
+    d.name = 'test.img'
+    return d
+
+
+@requires_fits
+@requires_data
+def test_xmmrgs_notice(make_data_path):
+    '''Test that notice and ignore works on XMMRGS dataset, which is
+    ordered in increasing wavelength, not energy'''
+    from sherpa.astro.ui.utils import Session
```

Comment:
I don't think we need the session class here - we should be able to

    from sherpa.astro.io import read_pha, read_rmf
    dat = read_pha(...)
    rmf = read_rmf(...)
    dat.set_rmf(rmf)   # or whatever the method is called
    
    dat.units = 'wave'
    dat.notice(18.8, 19.2)

    # don't need the session.get_data call
    assert len(dat.get_dep(filter=True)) == 41
    
    dat.ignore(10, 19)
    assert len(...) == 20

It might be good to check the output of `dat.get_filter(format='%.4f')` after both filters [adjust the format setting to something sensible for this data set].